File: examples/ruby/basic-bot.rb
require 'discordrb'
require 'dotenv'

Dotenv.load
token = ENV['TOKEN']
prefix = ENV['PREFIX'] || '!'

bot = Discordrb::Bot.new(token: token, intents: :all)

bot.ready do |event|
  puts "Logged in as #{bot.profile.username}"

  bot.register_application_command(:ping, 'Replies with Pong!')
  puts 'Slash command registered'
end

bot.application_command(:ping) do |event|
  event.respond(content: 'Pong!')
end

bot.message do |event|
  next if event.author.bot_account?

  content = event.content
  next unless content.start_with?(prefix)

  args = content[prefix.length..].split
  next if args.empty?

  if args[0].downcase == 'ping'
    event.respond('Pong!')
  end
end

bot.run

File: examples/php/basic-bot.php
<?php

require 'vendor/autoload.php';

use Discord\Discord;
use Discord\WebSockets\Event;
use Discord\Builders\MessageBuilder;

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->safeLoad();

$token = $_ENV['TOKEN'];
$prefix = $_ENV['PREFIX'] ?? '!';

$discord = new Discord(['token' => $token]);

$discord->on(Event::READY, function (Discord $discord) {
  echo "Logged in as {$discord->user->username}" . PHP_EOL;

  $command = new \Discord\Parts\Interactions\Command\Command($discord, [
    'name' => 'ping',
    'description' => 'Replies with Pong!',
    'type' => \Discord\Parts\Interactions\Command\Command::CHAT_INPUT,
  ]);

  $discord->application->commands->save($command)->done(function () {
    echo "Slash command registered" . PHP_EOL;
  });
});

$discord->on(Event::INTERACTION_CREATE, function ($interaction) {
  if ($interaction->type === \Discord\Parts\Interactions\Interaction::TYPE_APPLICATION_COMMAND) {
    if ($interaction->data->name === 'ping') {
      $interaction->respond(MessageBuilder::new()->setContent('Pong!'));
    }
  }
});

$discord->on(Event::MESSAGE_CREATE, function ($message) use ($discord, $prefix) {
  if ($message->author->bot) return;

  $content = $message->content;
  if (!str_starts_with($content, $prefix)) return;

  $args = explode(' ', substr($content, strlen($prefix)));
  if (empty($args[0])) return;

  if ($args[0] === 'ping') {
    $discord->client->sendMessage($message->channel_id, ['content' => 'Pong!']);
  }
});

$discord->run();
